lib/matplotlib/pyplot.py

```diff
@@ -1393,6 +1401,22 @@ def cla() -> None:
 
 ## More ways of creating Axes ##
 
+@overload
+def subplot(nrows: int, ncols: int, index: int, /, **kwargs): ...
+
+
+@overload
+def subplot(pos: int | SubplotSpec, /, **kwargs): ...
+
+
+@overload
+def subplot(ax: Axes, /): ...
```

Comment:
I don't see this form documented as acceptable, nor does it work.

Reply:
This was inferred from the docstring
````
call signatures:
    ...
    subplot(ax)
```
But indeed this does not work. It was intoduced in the docstring in #11459 and I suppose it was a false generalization because it also added `plt.add_axes(ax)`, which is valid. This and the corresponding docstring line should be removed.

Reply:
I have removed the overload and this point on the docstring. Not sure, but I think I have seen another methods with the same docstring error.